src/utils/sort-versions.ts

```diff
@@ -2,28 +2,54 @@ import * as semver from 'semver';
 
 import { RunnableVersion } from '../interfaces';
 
-function electronSemVerCompare(a: semver.SemVer, b: semver.SemVer) {
-  const l = a.compareMain(b);
-  if (l) return l;
-  // Electron's approach is nightly -> other prerelease tags -> stable,
-  // so force `nightly` to sort before other prerelease tags.
-  const [prea] = a.prerelease;
-  const [preb] = b.prerelease;
-  if (prea === 'nightly' && preb !== 'nightly') return -1;
-  if (prea !== 'nightly' && preb === 'nightly') return 1;
-  return a.comparePre(b);
-}
+const preTags = ['nightly', 'alpha', 'beta'];
 
-function compare(a: semver.SemVer | string, b: semver.SemVer | string) {
-  // non-semver goes first, sorted lexicographically
-  const astr = typeof a === 'string';
-  const bstr = typeof b === 'string';
-  if (astr && bstr) return (a as string).localeCompare(b as string);
-  if (astr) return -1;
-  if (bstr) return 1;
+/**
+ * Sorts prerelease tags such that nightly -> alpha -> beta.
+ *
+ * @param a Prerelease tag data for the old version.
+ * @param b Prerelease tag data for the new version.
+ * @returns 0 | 1 | -1
+ */
+const preCompare = (
+  a: readonly (string | number)[],
+  b: readonly (string | number)[],
+) => {
+  const one = preTags.indexOf(a[0] as string);
+  const two = preTags.indexOf(b[0] as string);
```

Comment:
It seems unsafe to array of strings *or* an array of numbers, but then to unconditionally treat it as an array of strings

Reply:
we can work around this if we'd like, but this was done because it's the required type for `Array.sort()`

Reply:
technically this is an array where the first index is a string and the rest are numbers - not sure the best way to type this tbh